Automattic/pocket-casts-ios
# Bookmarks: Add showing of podcast bookmarks

| 🎨 Design: [Figma](https://www.figma.com/file/Io7PCz1H1hmOgEE9eljoYW/Bookmarks?type=design&node-id=2033-33810&mode=dev) | 🛫 Depends on: https://github.com/Automattic/pocket-casts-ios/pull/1002 |
|:---:|:---:|


This adds showing of the podcast bookmarks in an overlay view. Since the designs don't quite handle for showing the list in an overlay I combined the look of the original designs, the episode list, and the profile bookmark designs. 

I also didn't see a "No results" view designed so I repurposed the empty state view to show this. 

The designs show, showing of the season and episode number but not all podcasts have this information so I've defaulted to showing the episode title for now. 

## Screenshots
| Default View | Searching | No Results |
|:---:|:---:|:---:|
|<img src="https://github.com/Automattic/pocket-casts-ios/assets/793774/87d97bfc-8218-443c-be22-43b99ddb8dc6" width="160" />|<img src="https://github.com/Automattic/pocket-casts-ios/assets/793774/592a81af-9f96-4e14-9055-25db7cbd4052" width="160" />|<img src="https://github.com/Automattic/pocket-casts-ios/assets/793774/47e10890-57d2-483a-bedf-0a093e26918e" width="160" />|

## Demo Video

https://github.com/Automattic/pocket-casts-ios/assets/793774/f3e7682c-15a9-4ae7-aae9-29a3b59a3116

## To test

1. Enable the Bookmarks Feature Flag by going to Profile > Cog > Beta Features > bookmarks
2. Open a podcast's details view for a podcast that has some bookmarks
3. Tap the Bookmarks item
4. ✅ Verify you see the bookmarks list
5. Tap the search bar
6. ✅ Verify the keyboard opens and you see the cancel item
7. Tap the cancel item
8. ✅ Verify the keyboard dismisses
9. Tap the search bar again
10. Enter some text that does not appear in your bookmarks
11. ✅ Verify you see the 'No Results' view
12. Tap the 'Clear Search' item in the no results view
13. ✅ Verify the search resets
14. Enter some text that only appears in some of the bookmarks
15. ✅ Verify the items are filtered
16. Press and hold on one of those items
17. Edit the title of the bookmark to something else
18. ✅ Verify that bookmark is then filtered out automatically
19. Tap the Cancel button
20. ✅ Verify all the bookmarks are displayed
21. Tap and hold to enter multi select
22. ✅ Verify the Select All and Cancel buttons appear in the top header
23. ✅ Verify they work
24. Tap Cancel
25. Tap the dismiss icon
26. ✅ Verify the view dismisses
27. Open the bookmarks list again
28. Tap the play button for an episode that is not currently playing
29. ✅ Verify the view dismisses, and the episode starts playing at the bookmark time
30. Open the bookmarks list again
31. Tap on the play button for an episode that is currently playing but at a different time
32. ✅ Verify the view dismisses and the episode jumps to that time

## Checklist
